app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,73 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
```

Comment:
Please remove this action. It is not a requirement for the project. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from Odin. You might wonder why it is important so I will explain. In your routes file, you have to use `:only` option for the specified routes defined in the screenshot. Therefore, the other  ones I am asking you to remove will be redundant.